### DISCUSSION TITLE
confusion in readme. Should I use fit()?

### DISCUSSION THREAD

**Discussion by feifeibear** [hpcaitech/ColossalAI#180] [Community | Q&A]
I saw the example code in README.md. It does not use the fit() API to train. However, the example in ColossalExample all use the fit(). Is it recommended to use fit() as the default training API? If it is, would it be better to present an example using fit() in the README.md?

**Comment by FrankLeeeee** [comment]
Hi, in the examples, we are preparing two versions of code in progress, one uses engine and one uses trainer, for example [resnet](https://github.com/hpcaitech/ColossalAI-Examples/tree/main/image/resnet). We treat trainer as high-level wrapper for engine to execute training, but engine is generally more flexible. Thus, we generally recommend using engine first as it is more compatible with the pytorch workflow, but trainer can come handy if you get familar with it.
